src/Linter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace SLLH\ComposerLint;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class Linter
+{
+    /**
+     * @param array $manifest composer.json file manifest
+     *
+     * @return string[]
+     */
+    public function validate($manifest)
+    {
+        $errors = array();
+
+        if (isset($manifest['config']['sort-packages']) && $manifest['config']['sort-packages']) {
+            foreach (array('require', 'require-dev', 'conflict', 'replace', 'provide', 'suggest') as $linksSection) {
+                if (array_key_exists($linksSection, $manifest) && !$this->packagesAreSorted($manifest[$linksSection])) {
+                    array_push($errors, 'Links under '.$linksSection.' section are not sorted.');
+                }
+            }
+        }
+
+        return $errors;
+    }
+
+    private function packagesAreSorted(array $packages)
+    {
+        $names = array_keys($packages);
+
+        $hasPHP = in_array('php', $names);
+        $extNames = array_filter($names, function ($name) {
+            return 'ext-' === substr($name, 0, 4) && !strstr($name, '/');
```

Comment:
This case will lead an error, some other not.

This is certainly why the yoda condition is on the SF coding standards.

And I approve this. I prefer put yoda condition everywhere and be sure to have no error instead of always asking "May this will raise an error or not?" on each comparison instead of being concentrate on the logic itself.